packages/devextreme/js/__internal/grids/data_grid/export/m_export.ts

```diff
@@ -608,11 +608,18 @@ export class ExportController extends dataGridCore.ViewController {
     return headersView && headersView.isVisible() ? headersView.getColumnWidths() : rowsView.getColumnWidths();
   }
 
-  init() {
-    if (this.option('export.enabled') && !isDefined(this.option('onExporting'))) {
+  private throwWarningIfNoOnExportingEvent(): void {
+    const hasProp = isDefined(this.option('onExporting'));
+    const hasEvent = (this.component as any)._eventsStrategy?.hasEvent('exporting');
```

Comment:
I think it's better to use the [hasActionSubscription](https://github.com/DevExpress/DevExtreme/blob/23_2/packages/devextreme/js/core/component.js#L353) method. It is available at the component level